subprojects/docs/src/snippets/buildCache/cacheable-bundle/kotlin/build.gradle.kts

```diff
@@ -39,7 +45,9 @@ tasks.register<NpmTask>("bundle") {
 
 tasks.register("printBundle") {
     dependsOn("bundle")
+
+    val bundleFile = file("$buildDir/bundle.js")
     doLast {
-        println(file("$buildDir/bundle.js").readText())
+        println(bundleFile.readText())
     }
 }
```

Comment:
When I run:

```
cd subprojects/docs/src/snippets/buildCache/cacheable-bundle/groovy
../../../../../../../gradlew printBundle
```

I get:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':bundle'.
> Multiple input file properties with name 'scripts'

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 408ms
```

which makes sense, as we are declaring `scripts` inputs in an [ad-hoc way](https://github.com/gradle/gradle/pull/21887/files#diff-4500d0398566fb3c898c63fbd68d2acd423bb82f4dcbc33819cc80b72d46cbadR33) in the configuration action, and again in the task implementation itself.

I am not sure why that problem is not reproduced when running `./gradlew :docs:docsTest --tests ...` as described in the GH issue.

I would suggest fixing the fragment that is included in the documentation - possibly putting inside the "bundle-task" fragment  the entire call to `tasks.register()` on [line 46](https://github.com/gradle/gradle/pull/21887/files#diff-4500d0398566fb3c898c63fbd68d2acd423bb82f4dcbc33819cc80b72d46cbadR46).

Reply:
After updating the soultion, I ran sample projects in standalone way as you said, they do pass now.

> Instead of making changes to "fake NpmTask", which is not included in the documentation, I would suggest fixing the fragment that is actually included in the documentation - possibly putting inside the "bundle-task" fragment the entire call to tasks.register() on [line 46](https://github.com/gradle/gradle/pull/21887/files#diff-4500d0398566fb3c898c63fbd68d2acd423bb82f4dcbc33819cc80b72d46cbadR46).

I think that the "public" fragment doesn't require fixing here - it doesn't violate any cc rules. The problems originate from the helping "fake" code.

> possibly putting inside the "bundle-task" fragment the entire call to tasks.register() on [line 46](https://github.com/gradle/gradle/pull/21887/files#diff-4500d0398566fb3c898c63fbd68d2acd423bb82f4dcbc33819cc80b72d46cbadR46).

It's not clear for me how it would help; it would make the public doc inconsistent with the [snippet right above this](https://docs.gradle.org/7.5.1/userguide/build_cache.html#using_annotations) and require correspondent changes in that snippet. And the [fix for that](https://github.com/gradle/gradle/pull/21869) didn't do it.